app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java

```diff
@@ -221,9 +235,10 @@ public void refreshCalendar(View v) {
                         }
                         calendarAdapter.notifyDataSetChanged();
                         calendar.setEvents(newEvents);
-                    }
-                    else {
-                        Toast.makeText(v.getContext(), v.getContext().getString(R.string.refresh_calendar_fail), Toast.LENGTH_SHORT).show();
+                    } else {
+                        logAndToast(Arrays.asList("MainScreenActivity",
+                                "refreshCalendar:failureToRefresh"), task.getException(),
+                                v.getContext(), v.getContext().getString(R.string.refresh_calendar_fail));
```

Comment:
Can these methods not be moved in their respective packages ? They dont feel like they belong in MainScreenActivity to me

Reply:
Good idea

Reply:
But apparently Code Climate is not okay with that.

Reply:
I had a similary problem last week, codeclimate only allows a max of 4 arguments per method. So it's difficult to move things around when you need a lot of outside information. Perhaps we could have some a controller or decorator for the calendar which receives many params from MainScreenActivity & uses those in its methods. Not sure if that's a good solution though

Reply:
For now I increased the limit to 7 but we can talk about it during our meeting.